src/core/layout/qgslayoutitemlabel.cpp

```diff
@@ -424,7 +452,31 @@ bool QgsLayoutItemLabel::readPropertiesFromElement( const QDomElement &itemElem,
   mVAlignment = static_cast< Qt::AlignmentFlag >( itemElem.attribute( QStringLiteral( "valign" ) ).toInt() );
 
   //font
-  QgsFontUtils::setFromXmlChildNode( mFont, itemElem, QStringLiteral( "LabelFont" ) );
+  QDomNodeList textFormatNodeList = itemElem.elementsByTagName( QStringLiteral( "text-style" ) );
+  if ( !textFormatNodeList.isEmpty() )
+  {
+    QDomElement textFormatElem = textFormatNodeList.at( 0 ).toElement();
+    mFormat.readXml( textFormatElem, context );
+  }
+  else
+  {
+    QFont f;
+    if ( !QgsFontUtils::setFromXmlChildNode( f, itemElem, QStringLiteral( "LabelFont" ) ) )
+    {
+      f.fromString( itemElem.attribute( QStringLiteral( "font" ), QString() ) );
+    }
+    mFormat.setFont( f );
+    if ( f.pointSizeF() > 0 )
+    {
+      mFormat.setSize( f.pointSizeF() );
+      mFormat.setSizeUnit( QgsUnitTypes::RenderPoints );
+    }
+    else if ( f.pixelSize() > 0 )
+    {
+      mFormat.setSize( f.pixelSize() );
+      mFormat.setSizeUnit( QgsUnitTypes::RenderPixels );
+    }
+  }
 
   //font color
   const QDomNodeList fontColorList = itemElem.elementsByTagName( QStringLiteral( "FontColor" ) );
```

Comment:
@nyalldawson , I think could be (should be?) moved into the above else { // no text-style } block, logic would be easier to understand.